bin/preprocess_datavzrd.py

```diff
@@ -65,6 +72,14 @@ def process_ann_cols(variant_df, colinfo_df, ann_name_column, ann_group_column,
     # check for duplicated variant TSV columns or column info.
     check_for_duplicates(variant_df.columns, "variant TSV file column names")
     check_for_duplicates(colinfo_df[ann_name_column], "colinfo TSV file annotation name column")
+
+    # Here we check for colinfo columns that are sample-specific and duplicate them based on the sample names
+    colinfo_df = match_patterns_and_duplicate(variant_df, colinfo_df, "read_depth", ann_name_column, ann_label_column)
```

Comment:
This function is repeatedly, it is probably better to call it once and have the loop through [read_depth, allele_fraction, format] solved within the function.
This could reduce duplicated code and improve readability.